cpp/src/lance/encodings/fixed_size_binary.cc

```diff
@@ -0,0 +1,124 @@
+//  Copyright 2022 Lance Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lance/encodings/fixed_size_binary.h"
+
+#include <arrow/builder.h>
+
+#include "lance/arrow/type.h"
+#include "lance/encodings/encoder.h"
+#include "lance/encodings/plain.h"
+
+namespace lance::encodings {
+
+FixedSizeBinaryEncoder::FixedSizeBinaryEncoder(
```

Comment:
This is called FixedSizeBinaryEncoder/Decoder but it handles both FixedSizeBinary and FixedSizeList. Would it be better to call it FixedSizeEncoder/Decoder or something like that?

Reply:
Good point. So this encoding is in contrast to `VarBinary`, (var-sized binary). 
Maybe this is another argument to support merging with PlainEncoding